reactor-core/src/main/java/reactor/core/publisher/SinksSpecs.java

```diff
@@ -29,7 +29,7 @@
 
 }
 
-final class SerializedManySink<T> implements Many<T>, Scannable {
+final class SerializedManySink<T> implements InternalManySink<T>, Scannable {
```

Comment:
one risk with the combination of default implementation at interface level + use of a wrapper for serialization of calls is that even if the underlying delegate has a specific `emitNext` implem, the wrapper will revert it to the default behavior.

it leads to a situation where the `emitXxx` methods are part of the interface API but are not supposed to be overridden, right? 

at the same time, we cannot enforce an implementation through the use of an abstract class rather than an interface, because our concrete implementations already must extend `Flux` or `Mono` 😢

that's a drawback I was trying to solve by giving more responsabilities to the `strategy`, so that the custom behavior would live in the strategy rather than the sink (and thus wrappers could reproduce the default behavior). I don't see that sort of composability here.

Reply:
> if the underlying delegate has a specific `emitNext` implem, the wrapper will revert it to the default behavior.

don't we assume that `emitNext` is an API _on top_ of `tryEmitNext`? I think this is an important assumption that otherwise will make a lot of things harder (think changing from `emitNext` to `tryEmitNext` for better control and having to deal with a _behaviour_ change)

Reply:
that assumption cannot be enforce by the compiler, so I would at least document it (eg. for people tempted to implement their own `Sinks.Many`)